File: designpatterns/src/com/renho/behavior/observer/ISubject.java
package com.renho.behavior.observer;

import java.util.ArrayList;
import java.util.List;

public interface ISubject {

	public List<IObserver> list = new ArrayList<IObserver>();
	public void add(IObserver observer);
	public void remove(IObserver observer);
	public void notifyOb();
	
}

File: ideaworkspace/designpatterns/src/main/scala/com/renho/behavior/observer/ISubject.scala
package com.renho.behavior.observer

import scala.collection.mutable.ListBuffer

/**
  * Created by renho on 2016/7/11.
  */
trait ISubject {
	var list = new ListBuffer[IObserver]
	def add(observer: IObserver)
	def remove(observer: IObserver)
	def notifyObserver()
}
